AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/globalfunction/control/GlobalFunctionRepository.java

```diff
@@ -71,4 +75,20 @@ private GlobalFunctionEntity findFunctionByName(String name) {
 		List<Object> result = tq.getResultList();
 		return result.isEmpty() ? null : (GlobalFunctionEntity) result.get(0);
 	}
+
+	public boolean isExistingId(Integer id) {
+		return findFunctionById(id) != null;
+	}
+
+	private GlobalFunctionEntity findFunctionById(Integer id) {
+		CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+		CriteriaQuery<Object> query = cb.createQuery();
+		Root<GlobalFunctionEntity> from = query.from(GlobalFunctionEntity.class);
+		CriteriaQuery<Object> select = query.select(from);
+		Predicate predicate = cb.equal(from.get("id"), id);
+		query.where(predicate);
+		TypedQuery<Object> tq = entityManager.createQuery(select);
+		List<Object> result = tq.getResultList();
```

Comment:
You could use a simple hql query here:
```java
final TypedQuery<GlobalFunctionEntity> query = entityManager.createQuery("from GlobalFunctionEntity g where g.id = :id);
query.setParameter("id", id);
return query.getResultList();
```
Or is there a reason to use the `CriteriaBuilder` ?